core/services/vrf/listener_v2.go

```diff
@@ -178,8 +178,22 @@ type listenerV2 struct {
 }
 
 // Start starts listenerV2.
-func (lsn *listenerV2) Start(context.Context) error {
+func (lsn *listenerV2) Start(ctx context.Context) error {
 	return lsn.StartOnce("VRFListenerV2", func() error {
+		// Check gas limit configuration
```

Comment:
I wonder if we can put this in `ServicesForSpec` in `delegate.go`? My reasoning: that method will get called when a node operator tries to create a new VRF job, and it's better to error out on job construction with a very clear message (i.e "Node gas limit setting should be increased to <blah>") to avoid running the VRF job with bad configuration.

Putting it here works too, but is a little less forcing and possibly not as visible, since we still try to run the job and fulfill requests, albeit in a possibly degraded way. We could still possibly legitimately fulfill requests with a bad config (depending on how far is deviates from on-chain max gas limit) but we'd still be in a brittle state.

Thoughts?

Reply:
Yeah I considered just not running at all if the check fails, but what do we do if the GetConfig call fails (because an RPC is down or whatever)? I guess we could start up in that case and fatally error if GetConfig succeeds and the check fails. Also not sure if it's standard to put IO / network calls inside ServicesForSpec